### PR TITLE
use planemo-ci-action for PR and CI workflow

### PR DESCRIPTION
**Pull request by bernt-matthias**
supersedes https://github.com/galaxyproject/tools-iuc/pull/3257
needs https://github.com/galaxyproject/planemo-ci-action/pull/1

I think I tested everything except for the deployment.

Co-authored-by: Marius van den Beek <m.vandenbeek@gmail.com>

FOR CONTRIBUTOR:
* [ ] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -50,43 +51,30 @@ jobs:
...
-      run: |
-        touch tool.xml
-        PIP_QUIET=1 planemo test --galaxy_python_version ${{ matrix.python-version }} --no_conda_auto_init --galaxy_source 'https://github.com/${{ steps.get-fork-branch.outputs.fork }}/galaxy' --galaxy_branch "${{ steps.get-fork-branch.outputs.branch }}" --galaxy_python_version ${{ matrix.python-version }}
     - uses: actions/checkout@v2
       with:
         fetch-depth: 1
-    - name: Planemo ci_find_repos
-      run: planemo ci_find_repos --exclude packages --exclude deprecated --exclude_from .tt_skip --output repository_list.txt
+    - name: Fake a Planemo run to update cache and determine commit range, repositories, and chunks
+      uses: bernt-matthias/planemo-ci-action@master
```
Comment (bernt-matthias) [inline .github/workflows/ci.yaml:65]:
```suggestion
      uses: galaxyproject/planemo-ci-action@master
```

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -121,38 +109,17 @@ jobs:
...
-      run: planemo merge_test_reports json_output/*.json tool_test_output.json
-    - name: Create tool_test_output.html
-      run: planemo test_reports tool_test_output.json --test_output tool_test_output.html
-    - name: Copy artifacts into place
-      run: |
-        mkdir upload
-        mv tool_test_output.json tool_test_output.html upload/
+        key: pip_cache_py_${{ matrix.python-version }}_gxy_${{ needs.setup.outputs.galaxy-head-sha }}
+    - name: Planemo test
+      uses: bernt-matthias/planemo-ci-action@master
```
Comment (bernt-matthias) [inline .github/workflows/ci.yaml:114]:
```suggestion
      uses: galaxyproject/planemo-ci-action@master
```

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -183,19 +150,13 @@ jobs:
...
+        key: pip_cache_py_${{ matrix.python-version }}_gxy_${{ needs.setup.outputs.galaxy-head-sha }}
     - name: Combine outputs
-      run: find artifacts/ -name tool_test_output.json -exec sh -c 'planemo merge_test_reports "$@" tool_test_output.json' sh {} +
-    - name: Create tool_test_output.html
-      run: planemo test_reports tool_test_output.json --test_output tool_test_output.html
-    - name: Copy artifacts into place
-      run: |
-        mkdir upload
-        mv tool_test_output.json tool_test_output.html upload/
+      uses: bernt-matthias/planemo-ci-action@master
```
Comment (bernt-matthias) [inline .github/workflows/ci.yaml:155]:
```suggestion
      uses: galaxyproject/planemo-ci-action@master
```

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -221,14 +177,13 @@ jobs:
...
 
           [1]: ${{ steps.vars.outputs.run-url }}
-    - name: Check status of combined outputs
-      run: |
-        if jq '.["tests"][]["data"]["status"]' upload/tool_test_output.json | grep -v "success"; then
-            echo "Unsuccessful tests found, inspect the 'All tool test results' artifact for details."
-            exit 1
-        fi
+    - name: Check outputs
+      uses: bernt-matthias/planemo-ci-action@master
```
Comment (bernt-matthias) [inline .github/workflows/ci.yaml:186]:
```suggestion
      uses: galaxyproject/planemo-ci-action@master
```

File: .github/workflows/pr.yaml [hunk 5]
```diff
@@ -44,82 +46,44 @@ jobs:
...
-      run: echo "COMMIT_RANGE=HEAD~.." >> $GITHUB_ENV
-    - id: get-commit-range
-      run: echo "::set-output name=commit_range::$COMMIT_RANGE"
-    - name: Planemo ci_find_repos for the commit range
-      run: planemo ci_find_repos --changed_in_commit_range $COMMIT_RANGE --exclude packages --exclude deprecated --exclude_from .tt_skip --output repository_list.txt
-    - name: Show repository list
-      run: cat repository_list.txt
-    - uses: actions/upload-artifact@v2
+    - name: Fake a Planemo run to update cache and determine commit range, repositories, and chunks
+      uses: bernt-matthias/planemo-ci-action@master
```
Comment (bernt-matthias) [inline .github/workflows/pr.yaml:64]:
```suggestion
      uses: galaxyproject/planemo-ci-action@master
```

File: .github/workflows/pr.yaml [hunk 6]
```diff
@@ -130,45 +94,29 @@ jobs:
...
-        while read -r TOOL; do
-            # Check if any changed repo dir is a substring of $TOOL
-            if ! echo $TOOL | grep -qf ../workflow_artifacts/repository_list.txt; then
-                echo "Tool $TOOL not in changed repositories list: .shed.yml file missing" >&2
-                exit 1
-            fi
-        done < tool_list.txt
+        key: pip_cache_py_${{ matrix.python-version }}_gxy_${{ needs.setup.outputs.galaxy-head-sha }}
+    - name: Planemo lint 
+      uses: bernt-matthias/planemo-ci-action@master
```
Comment (bernt-matthias) [inline .github/workflows/pr.yaml:108]:
```suggestion
      uses: galaxyproject/planemo-ci-action@master
```

File: .github/workflows/pr.yaml [hunk 7]
```diff
@@ -286,57 +227,29 @@ jobs:
...
-    - name: Create tool_test_output.html
-      run: planemo test_reports tool_test_output.json --test_output tool_test_output.html
-    - name: Copy artifacts into place
-      run: |
-        mkdir upload
-        mv tool_test_output.json tool_test_output.html upload/
+        key: planemo_cache_py_${{ matrix.python-version }}_gxy_${{ needs.setup.outputs.galaxy-head-sha }}
+
+    - name: Planemo test
+      uses: bernt-matthias/planemo-ci-action@master
```
Comment (bernt-matthias) [inline .github/workflows/pr.yaml:244]:
```suggestion
      uses: galaxyproject/planemo-ci-action@master
```

File: .github/workflows/pr.yaml [hunk 8]
```diff
@@ -367,29 +280,22 @@ jobs:
...
+        key: pip_cache_py_${{ matrix.python-version }}_gxy_${{ needs.setup.outputs.galaxy-head-sha }}
     - name: Combine outputs
-      run: find artifacts/ -name tool_test_output.json -exec sh -c 'planemo merge_test_reports "$@" tool_test_output.json' sh {} +
-    - name: Create tool_test_output.html
-      run: planemo test_reports tool_test_output.json --test_output tool_test_output.html
-    - name: Copy artifacts into place
-      run: |
-        mkdir upload
-        mv tool_test_output.json tool_test_output.html upload/
+      uses: bernt-matthias/planemo-ci-action@master
```
Comment (bernt-matthias) [inline .github/workflows/pr.yaml:285]:
```suggestion
      uses: galaxyproject/planemo-ci-action@master
```

File: .github/workflows/pr.yaml [hunk 9]
```diff
@@ -367,29 +280,22 @@ jobs:
...
         name: 'All tool test results'
         path: upload
-    - name: Check status of combined outputs
-      run: |
-        if jq '.["tests"][]["data"]["status"]' upload/tool_test_output.json | grep -v "success"; then
-            echo "Unsuccessful tests found, inspect the 'All tool test results' artifact for details."
-            exit 1
-        fi
+    - name: Check outputs
+      uses: bernt-matthias/planemo-ci-action@master
```
Comment (bernt-matthias) [inline .github/workflows/pr.yaml:295]:
```suggestion
      uses: galaxyproject/planemo-ci-action@master
```

File: .github/workflows/pr.yaml [hunk 10]
```diff
@@ -403,34 +309,28 @@ jobs:
...
+#     - name: Install Planemo
+#       run: pip install planemo
     - name: Deploy on testtoolshed
-      env:
-        SHED_KEY: ${{ secrets.TTS_API_KEY }}
-      run: |
-        while read -r DIR; do
-            planemo shed_update --shed_target testtoolshed --shed_key "${{ env.SHED_KEY }}" --force_repository_creation "$DIR" || exit 1;
-        done < ../workflow_artifacts/repository_list.txt
+      uses: bernt-matthias/planemo-ci-action@master
```
Comment (bernt-matthias) [inline .github/workflows/pr.yaml:325]:
```suggestion
      uses: galaxyproject/planemo-ci-action@master
```

File: .github/workflows/pr.yaml [hunk 11]
```diff
@@ -403,34 +309,28 @@ jobs:
...
+        shed-key: ${{ secrets.TTS_API_KEY }}
       continue-on-error: true
     - name: Deploy on toolshed
-      env:
-        SHED_KEY: ${{ secrets.TS_API_KEY }}
-      run: |
-        while read -r DIR; do
-            planemo shed_update --shed_target toolshed --shed_key "${{ env.SHED_KEY }}" --force_repository_creation "$DIR" || exit 1;
-        done < ../workflow_artifacts/repository_list.txt
+      uses: bernt-matthias/planemo-ci-action@master
```
Comment (bernt-matthias) [inline .github/workflows/pr.yaml:332]:
```suggestion
      uses: galaxyproject/planemo-ci-action@master
```

File: .github/workflows/pr.yaml [hunk 12]
```diff
@@ -403,34 +309,28 @@ jobs:
     steps:
     - uses: actions/checkout@v2
       with:
-        fetch-depth: 1
+        fetch-depth: 0
```
Comment (bernt-matthias) [inline .github/workflows/pr.yaml:312]:
```suggestion
        fetch-depth: 1
```

File: .github/workflows/pr.yaml [hunk 13]
```diff
@@ -16,10 +16,12 @@ jobs:
     name: Setup cache and determine changed repositories
     runs-on: ubuntu-latest
     outputs:
-      galaxy_head_sha: ${{ steps.get-galaxy-sha.outputs.galaxy_head_sha }}
-      commit_range: ${{ steps.get-commit-range.outputs.commit_range }}
-      nchunks: ${{ steps.get-chunks.outputs.nchunks }}
-      chunk_list: ${{ steps.get-chunks.outputs.chunk_list }}
+      galaxy-head-sha: ${{ steps.get-galaxy-sha.outputs.galaxy-head-sha }}
+      commit-range: ${{ steps.discover.outputs.commit-range }}
+      repositories: ${{ steps.discover.outputs.repositories }}
```
Comment (bernt-matthias) [inline .github/workflows/pr.yaml:21]: Rename repositories and tools to repository-list and tool-list

File: .github/workflows/pr.yaml [hunk 14]
```diff
@@ -407,30 +313,24 @@ jobs:
...
     - name: Cache .cache/pip
       uses: actions/cache@v2
       id: cache-pip
       with:
         path: ~/.cache/pip
-        key: pip_cache_py_${{ matrix.python-version }}_gxy_${{ needs.setup.outputs.galaxy_head_sha }}
-    - name: Install Planemo
-      run: pip install planemo
+        key: pip_cache_py_${{ matrix.python-version }}_gxy_${{ needs.setup.outputs.galaxy-head-sha }}
+#     - name: Install Planemo
```
Comment (bgruening) [inline .github/workflows/pr.yaml:322]: this could be removed now

**Review by bgruening — APPROVED** [review]
pretty cool!

**Comment by bernt-matthias** [comment]
Note that we need https://github.com/galaxyproject/planemo-ci-action/pull/3 before merging this.

File: .github/workflows/pr.yaml (hunk 14 above)
Comment (bernt-matthias) [reply .github/workflows/pr.yaml]: done

**Comment by bernt-matthias** [comment]
Should be good to go, or @mvdbeek? .. I guess we should monitor the first runs of the workflows. Hope that this does not create to much noise, but the action should be quite well tested.

**Pull request merged**